bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/ContextInformationPopup.java

```diff
@@ -876,30 +876,24 @@ private void validateContextInformation() {
 		 */
 		if (!isValid(fContextInfoPopup))
 			return;
-
-		fContextInfoPopup.getDisplay().asyncExec(new Runnable() {
-
-			private ContextFrame fFrame= fContextFrameStack.peek();
```

Comment:
A similar pattern here: the frame is set once at instantiation for some reason, not on the execution. This looks like an actual intent to capture the "initial value before execution".